contracts/eosio.system/include/eosio.system/peer_keys.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <eosio/contract.hpp>
+#include <eosio/crypto.hpp>
+#include <eosio/name.hpp>
+
+#include <string>
+#include <optional>
+
+namespace eosiosystem {
+
+using eosio::name;
+using eosio::public_key;
+
+// -------------------------------------------------------------------------------------------------
+// -------------------------------------------------------------------------------------------------
+struct [[eosio::table("peerkeys"), eosio::contract("eosio.system")]] peer_key {
+   struct v0_data {
+      std::optional<public_key> pubkey; // peer key for network message authentication
```

Comment:
This is no longer `optional`.

Reply:
Doesn't `v0_data` in the `variant` provide that capability. We would never just add to `v0_data`, right?

Reply:
I think we could add a binary_extension to `v0_data` of we wanted to.